src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/entity/CertificateDB.kt

```diff
@@ -69,4 +70,11 @@ class CertificateDB(
     @Column(name = "uploader")
     var uploader: String?,
 
+    @Column(name = "document_id", unique = true)
+    val documentID: UUID = UUID.randomUUID(),
+
+    @OneToOne(cascade = [CascadeType.ALL], orphanRemoval = true)
```

Comment:
This way now queries on the certificate table are faster since it doesn't need to load also the file information. However, when fetching a certificate you now always also fetch the document file. Since you said you may want to fetch only the certificate header information without the file, this can slow you down. You can improve this by making the oneToOne relation lazy here

Reply:
Agree, adapted fetchType to the Lazy.